### ISSUE TITLE
dotnet pack includes analyzers in nuspec dependencies

### ISSUE THREAD

**Issue by skolima** [NuGet/Home#7051]
## Details about Problem

NuGet product used : dotnet.exe

dotnet.exe --version (if appropriate): 2.1.300

OS version (i.e. win10 v1607 (14393.321)): win10 v1803 (17134.112)

Worked before? If so, with which NuGet version: don't think so, at least not in any dotnet 2.0 builds (didn't try earlier)

## Detailed repro steps so we can see the same problem

1. `dotnet new classlib`

2. `dotnet add package Stylecop.Analyzers`

3. `dotnet pack`

The resulting `.nupkg` / `.nuspec` has a dependency on the analyzers package which is only a development time dependency. Same happens for other packages, e.g. `xunit.analyzers`, so it seems to be an issue with how the `.nuspec` is generated.

### Verbose Logs

Please include verbose logs (NuGet.exe <COMMAND> -verbosity detailed | dotnet.exe <COMMAND> --verbosity diag | etc...)
[dotnet.pack.log](https://github.com/NuGet/Home/files/2128108/dotnet.pack.log)

### Sample Project

Please see the above repro steps - no other code is needed.

**Comment by rohit21agrawal** [comment]
can you show your csproj after you do ``` dotnet add package Stylecop.Analyzers ``` . There was recently work done to persist the devDependency information to csproj , which the pack command should respect.

I think that feature is available in 2.1.400 version of SDK, so it will be fixed if you upgrade to that.

**Comment by skolima** [comment]
With 2.1.300:

```
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <TargetFramework>netstandard2.0</TargetFramework>
  </PropertyGroup>
  <ItemGroup>
    <PackageReference Include="Stylecop.Analyzers" Version="1.0.2" />
  </ItemGroup>
</Project>
```

With 2.1.400-preview-009053 (which doesn't include analyzers as dependency in the packaged nuspec):

```
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <TargetFramework>netstandard2.0</TargetFramework>
  </PropertyGroup>
  <ItemGroup>
    <PackageReference Include="Stylecop.Analyzers" Version="1.0.2">
      <IncludeAssets>runtime; build; native; contentfiles; analyzers</IncludeAssets>
      <PrivateAssets>all</PrivateAssets>
    </PackageReference>
  </ItemGroup>
</Project>
```

**Comment by skolima** [comment]
I have verified that with the `<PackageReference/>` node that 2.1.400 creates, packaging with 2.1.300 also creates expected nuspec dependencies (so, excludes Analyzers).

**Comment by rohit21agrawal** [comment]
glad it worked for you, please close the issue if this has been resolved for you. Thanks!

**Comment by rrelyea** [comment]
Fixed with devdependency work in 15.8 (closing as dupe)
